src/uct/ib/dc/dc_mlx5.c

```diff
@@ -954,11 +954,11 @@ static void uct_dc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface,
                                    UCS_MASK(UCT_IB_QPN_ORDER);
     uint8_t              dci     = uct_dc_mlx5_iface_dci_find(iface, qp_num);
     uct_rc_txqp_t        *txqp   = &iface->tx.dcis[dci].txqp;
-    uct_dc_mlx5_ep_t     *ep     = uct_dc_mlx5_ep_from_dci(iface, dci);
+    uct_dc_mlx5_ep_t     *ep     = iface->tx.dcis[dci].ep;
     ucs_status_t         ep_status;
     int16_t              outstanding;
 
-    if (!ep) {
+    if (!ep || uct_dc_mlx5_iface_is_dci_rand(iface)) {
```

Comment:
need to avoid access to iface->tx.dcis[dci].ep before we check the policy is rand